docs/developer/meta/config-models.md

```diff
@@ -56,46 +53,55 @@ def initialize_<ID>(values: dict[str, Any], **kwargs) -> dict[str, Any]:
 If such a validator exists in `validators.py`, then it is called once with the raw config that was supplied by the user.
 The returned mapping is used as the input config for the subsequent stages.
 
-### Default value population
+### Field
 
-If a field was not supplied by the user nor during the initialization stage, then its default value is
-taken from `defaults.py`. This stage is skipped for required fields.
+The value of each field goes through the following steps.
 
-### Default field validators
+#### Default value population
 
-At this point `pydantic` will parse the values and perform validation of types, etc.
+If a field was not supplied by the user nor during the [initialization stage](#initial), then its default value is
+taken from `defaults.py`. This stage is skipped for required fields.
 
-### Custom field validators
+#### Custom field validators
 
 The contents of `validators.py` are entirely custom and contain functions to perform extra validation if necessary.
 
 ```python
-def <ID>_<OPTION_NAME>(value: Any, *, field: pydantic.fields.ModelField, **kwargs) -> Any:
+def <ID>_<OPTION_NAME>(value: Any, *, field: pydantic.fields.FieldInfo, **kwargs) -> Any:
     ...
 ```
 
-Such validators are called for the appropriate field of the proper model if the option was supplied by the user.
+Such validators are called for the appropriate field of the proper model. The returned value is used as the
+new value of the option for the subsequent stages.
 
-The returned value is used as the new value of the option for the subsequent stages.
+!!! note
+    This only occurs if the option was supplied by the user.
 
-### Pre-defined field validators
+#### Pre-defined field validators
 
-A new `validators` key under the [value](https://datadoghq.dev/integrations-core/meta/config-specs/#values) property of config
+A `validators` key under the [value](https://datadoghq.dev/integrations-core/meta/config-specs/#values) property of config
 spec options is considered. Every entry will refer to a relative import path to a [field validator](#custom-field-validators)
 under `datadog_checks.base.utils.models.validation` and is executed in the defined order.
 
-The last returned value is used as the new value of the option for the [final](#final) stage.
+!!! note
+    This only occurs if the option was supplied by the user.
+
+#### Conversion to immutable types
+
+Every `list` is converted to `tuple` and every `dict` is converted to [types.MappingProxyType](https://docs.python.org/3/library/types.html#types.MappingProxyType).
+
+!!! note
+    A field or nested field would only be a `dict` when it is defined as a mapping with arbitrary keys. Otherwise, it would be a model with its own properties as usual.
```

Comment:
```suggestion
**Note**: A field or nested field would only be a `dict` when it is defined as a mapping with arbitrary keys. Otherwise, it would be a model with its own properties as usual.
```